### PR TITLE
Spring4 EOL

### PR DESCRIPTION
**Pull request by stevenschlansker**

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
   - add support for Moshi JSON mapping (#1809, thanks unoexperto!)
   - Register more array types like `boolean` out of the box, #1802
   - add Codec (combination of ArgumentFactory and ColumnMapper) to provide one API for serialization/deserialization.
+  - Deprecate Spring 4 support due to upstream EOL (this will be the last release!) Please move to Spring 5
```
Comment (leaumar) [inline RELEASE_NOTES.md:9]: strange sentence construction: punctuation on a parenthesis but then a missing period

File: spring5/src/main/java/org/jdbi/v3/spring5/JdbiUtil.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        }
+
+        @Override
+        public void suspend() {
+            TransactionSynchronizationManager.unbindResource(db);
+        }
+
+        @Override
+        public void beforeCompletion() {
+            TRANSACTIONAL_HANDLES.remove(handle);
```
Comment (leaumar) [inline spring5/src/main/java/org/jdbi/v3/spring5/JdbiUtil.java:86]: I'm not particularly familiar with this advanced spring plumbing, just wanna ask if we're sure this collection doesn't need to be particularly threading-proof?

File: spring5/src/test/java/org/jdbi/v3/spring5/TestJdbiFactoryBean.java [hunk 3]
```diff
@@ -13,13 +13,137 @@
...
+    @Test
+    public void testFailsViaException() {
+        assertThatExceptionOfType(ForceRollback.class).isThrownBy(() -> {
+            service.inPropagationRequired(aJdbi -> {
+                final Handle h = JdbiUtil.getHandle(aJdbi);
+                final int count = h.execute("insert into something (id, name) values (7, 'ignored')");
+                if (count == 1) {
+                    throw new ForceRollback();
+                } else {
+                    throw new RuntimeException("!ZABAK");
```
Comment (leaumar) [inline spring5/src/test/java/org/jdbi/v3/spring5/TestJdbiFactoryBean.java:70]: what's this a reference to?

File: spring5/src/test/java/org/jdbi/v3/spring5/TestJdbiFactoryBean.java [hunk 4]
```diff
@@ -13,13 +13,137 @@
...
+    }
+
+    @Test
+    public void testGlobalDefinedAttribute() {
+        assertThat(jdbi.getConfig(SqlStatements.class).getAttribute("foo"))
+                .isEqualTo("bar"); // see test-context.xml
+    }
+
+    @Test
+    public void testNested() {
```
Comment (leaumar) [inline spring5/src/test/java/org/jdbi/v3/spring5/TestJdbiFactoryBean.java:100]: github doesn't play well with these big diffs, I take it no meaningful content was changed?

**Review by leaumar — APPROVED** [review]

File: spring5/src/main/java/org/jdbi/v3/spring5/JdbiUtil.java (hunk 2 above)
Comment (stevenschlansker) [reply spring5/src/main/java/org/jdbi/v3/spring5/JdbiUtil.java:86]: That's a good point, this sure seems to be not thread safe.  But we should make functional changes in a separate PR to just reorganizing.

File: spring5/src/test/java/org/jdbi/v3/spring5/TestJdbiFactoryBean.java (hunk 3 above)
Comment (stevenschlansker) [reply spring5/src/test/java/org/jdbi/v3/spring5/TestJdbiFactoryBean.java:70]: I have no idea, I just moved the code.

File: spring5/src/test/java/org/jdbi/v3/spring5/TestJdbiFactoryBean.java (hunk 4 above)
Comment (stevenschlansker) [reply spring5/src/test/java/org/jdbi/v3/spring5/TestJdbiFactoryBean.java:100]: Only `spring4` -> `spring5` updates, no functional changes.

**Pull request merged**